lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1377,125 +1377,98 @@ def _parse_detail_errors(self, error: Any) -> List[str]:
 
     # the VM may not be queried after deployed. use retry to mitigate it.
     @retry(exceptions=LisaException, tries=150, delay=2)
-    def _load_vms(
-        self, environment: Environment, log: Logger
-    ) -> Dict[str, VirtualMachine]:
+    def load_vms_info(
+        self, resource_group_name: str, log: Logger
+    ) -> Tuple[Dict[str, VirtualMachine], Dict[str, NetworkInterface], Dict[str, str]]:
         compute_client = get_compute_client(self, api_version="2020-06-01")
-        environment_context = get_environment_context(environment=environment)
+        network_client = get_network_client(self)
 
-        log.debug(
-            f"listing vm in resource group "
-            f"'{environment_context.resource_group_name}'"
-        )
+        log.debug(f"listing vm in resource group {resource_group_name}")
         vms_map: Dict[str, VirtualMachine] = {}
-        vms = compute_client.virtual_machines.list(
-            environment_context.resource_group_name
-        )
+        vms = compute_client.virtual_machines.list(resource_group_name)
         for vm in vms:
             vms_map[vm.name] = vm
             log.debug(f"  found vm {vm.name}")
         if not vms_map:
             raise LisaException(
-                f"deployment succeeded, but VM not found in 5 minutes "
-                f"from '{environment_context.resource_group_name}'"
+                "deployment succeeded, but VM not found in 5 minutes "
+                f"from '{resource_group_name}'"
             )
-        return vms_map
-
-    # Use Exception, because there may be credential conflict error. Make it
-    # retriable.
-    @retry(exceptions=Exception, tries=150, delay=2)
-    def _load_nics(
-        self, environment: Environment, log: Logger
-    ) -> Dict[str, NetworkInterface]:
-        network_client = get_network_client(self)
-        environment_context = get_environment_context(environment=environment)
 
-        log.debug(
-            f"listing network interfaces in resource group "
-            f"'{environment_context.resource_group_name}'"
-        )
-        # load nics
         nics_map: Dict[str, NetworkInterface] = {}
-        network_interfaces = network_client.network_interfaces.list(
-            environment_context.resource_group_name
-        )
-        for nic in network_interfaces:
-            # nic name is like lisa-test-20220316-182126-985-e0-n0-nic-2, get vm
-            # name part for later pick only find primary nic, which is ended by
-            # -nic-0
-            node_name_from_nic = RESOURCE_ID_NIC_PATTERN.findall(nic.name)
```

Comment:
Remove the patterns, if they are not used now.

Reply:
removed